modules/internal/String.chpl

```diff
@@ -1561,6 +1561,43 @@ module String {
       return _join(S);
     }
 
+    /*
+      Same as the varargs version, but with all the strings in a list.
+
+      :arg listToJoin: list contaning elements whose string values will be concatenated
+
+      :returns: A new string, which is the concatenation of all of the strings
+      passed in with the receiving string inserted between them.
+
+      .. code-block:: chapel
+
+          var someList = new list(string);
+          someList.append("a");
+          someList.append("10");
+          someList.append("d");
+          var x = "|".join(someList);
+          writeln(x); //prints "a|10|d"
+    
+    */
+
+    proc join(listToJoin: list(string)) : string {
+      return _join(listToJoin);
+    }
+
+    pragma "no doc"
+    proc _join(listToJoin: list(string)) : string where isList(listToJoin) {
```

Comment:
Instead of a private method in a where clause, you can just move the code in this routine up into `proc join(listToJoin: list(string)): string;`.